### ISSUE TITLE
Quotes around str parameters

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-jsonpreprocessor#80]
Defined in JSON file is:

```
"val10" : true,
"val11" : "true",
"val12" : false,
"val13" : "false",
"val14" : null,
"val15" : "null"
```

JsonPreprocessor returns:

```
'val10': True,
'val11': '"true"',
'val12': False,
'val13': '"false"',
'val14': None,
'val15': 'null'
```

The value of string parameters `val11` and `val13` contain the quotes that are used to define the strings in JSON file.

But expected is:

```
'val11': 'true',
'val13': 'false',
```

**Comment by HolQue** [comment]
Extended version:

Defined in JSON:

```
{
   "param_01" : "string",
   "param_02" : 123,
   "param_03" : "123",
   "param_04" : 4.56,
   "param_05" : "4.56",
   "param_06" : true,
   "param_07" : "true",
   "param_08" : false,
   "param_09" : "false",
   "param_10" : null,
   "param_11" : "null",
   "param_12" : True,
   "param_13" : "True",
   "param_14" : False,
   "param_15" : "False",
   "param_16" : None,
   "param_17" : "None",
   //
   "param_18" : ${param_01},
   "param_19" : "${param_01}",
   "param_20" : ${param_02},
   "param_21" : "${param_02}",
   "param_22" : ${param_03},
   "param_23" : "${param_03}",
   "param_24" : ${param_04},
   "param_25" : "${param_04}",
   "param_26" : ${param_05},
   "param_27" : "${param_05}",
   "param_28" : ${param_06},
   "param_29" : "${param_06}",
   "param_30" : ${param_07},
   "param_31" : "${param_07}",
   "param_32" : ${param_08},
   "param_33" : "${param_08}",
   "param_34" : ${param_09},
   "param_35" : "${param_09}",
   "param_36" : ${param_10},
   "param_37" : "${param_10}",
   "param_38" : ${param_11},
   "param_39" : "${param_11}",
   "param_40" : ${param_12},
   "param_41" : "${param_12}",
   "param_42" : ${param_13},
   "param_43" : "${param_13}",
   "param_44" : ${param_14},
   "param_45" : "${param_14}",
   "param_46" : ${param_15},
   "param_47" : "${param_15}",
   "param_48" : ${param_16},
   "param_49" : "${param_16}",
   "param_50" : ${param_17},
   "param_51" : "${param_17}"
}
```

This file is imported in another file that defines a list of all parameters above:

```
   // taking over the already defined basic data types
   "[import]" : "./jpp-test_config_0002.jsonp",
   //
   "param_100" : [${param_01},
                  ${param_02},
                  ${param_03},
                  .....
                  ${param_51}]
```

JsonPreprocessor returns:

```
[DICT] (52/1) > {param_01} [STR]  :  'string'
[DICT] (52/2) > {param_02} [INT]  :  123
[DICT] (52/3) > {param_03} [STR]  :  '123'
[DICT] (52/4) > {param_04} [FLOAT]  :  4.56
[DICT] (52/5) > {param_05} [STR]  :  '4.56'
[DICT] (52/6) > {param_06} [BOOL]  :  True
[DICT] (52/7) > {param_07} [STR]  :  '"true"'
[DICT] (52/8) > {param_08} [BOOL]  :  False
[DICT] (52/9) > {param_09} [STR]  :  '"false"'
[DICT] (52/10) > {param_10} [NONE]  :  None
[DICT] (52/11) > {param_11} [STR]  :  'null'
[DICT] (52/12) > {param_12} [BOOL]  :  True
[DICT] (52/13) > {param_13} [STR]  :  '"True"'
[DICT] (52/14) > {param_14} [BOOL]  :  False
[DICT] (52/15) > {param_15} [STR]  :  '"False"'
[DICT] (52/16) > {param_16} [NONE]  :  None
[DICT] (52/17) > {param_17} [STR]  :  '"None"'
[DICT] (52/18) > {param_18} [STR]  :  'string'
[DICT] (52/19) > {param_19} [STR]  :  'string'
[DICT] (52/20) > {param_20} [INT]  :  123
[DICT] (52/21) > {param_21} [STR]  :  '123'
[DICT] (52/22) > {param_22} [STR]  :  '123'
[DICT] (52/23) > {param_23} [STR]  :  '123'
[DICT] (52/24) > {param_24} [FLOAT]  :  4.56
[DICT] (52/25) > {param_25} [STR]  :  '4.56'
[DICT] (52/26) > {param_26} [STR]  :  '4.56'
[DICT] (52/27) > {param_27} [STR]  :  '4.56'
[DICT] (52/28) > {param_28} [BOOL]  :  True
[DICT] (52/29) > {param_29} [STR]  :  '"True"'
[DICT] (52/30) > {param_30} [STR]  :  '"true"'
[DICT] (52/31) > {param_31} [STR]  :  '"true"'
[DICT] (52/32) > {param_32} [BOOL]  :  False
[DICT] (52/33) > {param_33} [STR]  :  '"False"'
[DICT] (52/34) > {param_34} [STR]  :  '"false"'
[DICT] (52/35) > {param_35} [STR]  :  '"false"'
[DICT] (52/36) > {param_36} [NONE]  :  None
[DICT] (52/37) > {param_37} [STR]  :  '"None"'
[DICT] (52/38) > {param_38} [STR]  :  'null'
[DICT] (52/39) > {param_39} [STR]  :  'null'
[DICT] (52/40) > {param_40} [BOOL]  :  True
[DICT] (52/41) > {param_41} [STR]  :  '"True"'
[DICT] (52/42) > {param_42} [STR]  :  '"True"'
[DICT] (52/43) > {param_43} [STR]  :  '"True"'
[DICT] (52/44) > {param_44} [BOOL]  :  False
[DICT] (52/45) > {param_45} [STR]  :  '"False"'
[DICT] (52/46) > {param_46} [STR]  :  '"False"'
[DICT] (52/47) > {param_47} [STR]  :  '"False"'
[DICT] (52/48) > {param_48} [NONE]  :  None
[DICT] (52/49) > {param_49} [STR]  :  '"None"'
[DICT] (52/50) > {param_50} [STR]  :  '"None"'
[DICT] (52/51) > {param_51} [STR]  :  '"None"'
[DICT] (52/52) > {param_100} [LIST] (51/1) > [STR]  :  'string'
[DICT] (52/52) > {param_100} [LIST] (51/2) > [INT]  :  123
[DICT] (52/52) > {param_100} [LIST] (51/3) > [STR]  :  '123'
[DICT] (52/52) > {param_100} [LIST] (51/4) > [FLOAT]  :  4.56
[DICT] (52/52) > {param_100} [LIST] (51/5) > [STR]  :  '4.56'
[DICT] (52/52) > {param_100} [LIST] (51/6) > [BOOL]  :  True
[DICT] (52/52) > {param_100} [LIST] (51/7) > [STR]  :  '"true"'
[DICT] (52/52) > {param_100} [LIST] (51/8) > [BOOL]  :  False
[DICT] (52/52) > {param_100} [LIST] (51/9) > [STR]  :  '"false"'
[DICT] (52/52) > {param_100} [LIST] (51/10) > [NONE]  :  None
[DICT] (52/52) > {param_100} [LIST] (51/11) > [STR]  :  'null'
[DICT] (52/52) > {param_100} [LIST] (51/12) > [BOOL]  :  True
[DICT] (52/52) > {param_100} [LIST] (51/13) > [STR]  :  '"True"'
[DICT] (52/52) > {param_100} [LIST] (51/14) > [BOOL]  :  False
[DICT] (52/52) > {param_100} [LIST] (51/15) > [STR]  :  '"False"'
[DICT] (52/52) > {param_100} [LIST] (51/16) > [NONE]  :  None
[DICT] (52/52) > {param_100} [LIST] (51/17) > [STR]  :  '"None"'
[DICT] (52/52) > {param_100} [LIST] (51/18) > [STR]  :  '${param_01}'
[DICT] (52/52) > {param_100} [LIST] (51/19) > [STR]  :  'str(${param_01})'
[DICT] (52/52) > {param_100} [LIST] (51/20) > [STR]  :  '${param_02}'
[DICT] (52/52) > {param_100} [LIST] (51/21) > [STR]  :  'str(${param_02})'
[DICT] (52/52) > {param_100} [LIST] (51/22) > [STR]  :  '${param_03}'
[DICT] (52/52) > {param_100} [LIST] (51/23) > [STR]  :  'str(${param_03})'
[DICT] (52/52) > {param_100} [LIST] (51/24) > [STR]  :  '${param_04}'
[DICT] (52/52) > {param_100} [LIST] (51/25) > [STR]  :  'str(${param_04})'
[DICT] (52/52) > {param_100} [LIST] (51/26) > [STR]  :  '${param_05}'
[DICT] (52/52) > {param_100} [LIST] (51/27) > [STR]  :  'str(${param_05})'
[DICT] (52/52) > {param_100} [LIST] (51/28) > [STR]  :  '${param_06}'
[DICT] (52/52) > {param_100} [LIST] (51/29) > [STR]  :  'str(${param_06})'
[DICT] (52/52) > {param_100} [LIST] (51/30) > [STR]  :  '${param_07}'
[DICT] (52/52) > {param_100} [LIST] (51/31) > [STR]  :  'str(${param_07})'
[DICT] (52/52) > {param_100} [LIST] (51/32) > [STR]  :  '${param_08}'
[DICT] (52/52) > {param_100} [LIST] (51/33) > [STR]  :  'str(${param_08})'
[DICT] (52/52) > {param_100} [LIST] (51/34) > [STR]  :  '${param_09}'
[DICT] (52/52) > {param_100} [LIST] (51/35) > [STR]  :  'str(${param_09})'
[DICT] (52/52) > {param_100} [LIST] (51/36) > [STR]  :  '${param_10}'
[DICT] (52/52) > {param_100} [LIST] (51/37) > [STR]  :  'str(${param_10})'
[DICT] (52/52) > {param_100} [LIST] (51/38) > [STR]  :  '${param_11}'
[DICT] (52/52) > {param_100} [LIST] (51/39) > [STR]  :  'str(${param_11})'
[DICT] (52/52) > {param_100} [LIST] (51/40) > [STR]  :  '${param_12}'
[DICT] (52/52) > {param_100} [LIST] (51/41) > [STR]  :  'str(${param_12})'
[DICT] (52/52) > {param_100} [LIST] (51/42) > [STR]  :  '${param_13}'
[DICT] (52/52) > {param_100} [LIST] (51/43) > [STR]  :  'str(${param_13})'
[DICT] (52/52) > {param_100} [LIST] (51/44) > [STR]  :  '${param_14}'
[DICT] (52/52) > {param_100} [LIST] (51/45) > [STR]  :  'str(${param_14})'
[DICT] (52/52) > {param_100} [LIST] (51/46) > [STR]  :  '${param_15}'
[DICT] (52/52) > {param_100} [LIST] (51/47) > [STR]  :  'str(${param_15})'
[DICT] (52/52) > {param_100} [LIST] (51/48) > [STR]  :  '${param_16}'
[DICT] (52/52) > {param_100} [LIST] (51/49) > [STR]  :  'str(${param_16})'
[DICT] (52/52) > {param_100} [LIST] (51/50) > [STR]  :  '${param_17}'
[DICT] (52/52) > {param_100} [LIST] (51/51) > [STR]  :  'str(${param_17})'
```

The resolving of values do not work properly.

**Comment by test-fullautomation** [comment]
Hi Son,
please look into this.
Expectation is what Holger writes.
Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Hogler, Hello Thomas,

In Holger example, I observe a nested parameter is used to other nested parameter, for example:

-  The parameter param_02 is defined with value 123 - "param_02" : 123,
-  The first nested parameter param_20 is defined with value ${param_02} - "param_20" : ${param_02},
-  Then parameter param_20 again is used as nested parameter in param_100[..., ${param_20}, ...]

My concern: Dose this scenario occurs in reality? Because user can easy to set nested parameter ${param_02} in param_100[....] instead.

Thank you,
Son

**Comment by HolQue** [comment]
Hi Son,

let's say, a user implements:

```
a = 1
b = a
print(b)
```

Why shall such a scenario not occur in reality?

When I apply your concern to this example, then your question would be:

Why is the implementation like that: `print(b)`, when also `print(a)` would print the value?

In general it should be possible to use every existing variable inside code. Why not?

**Comment by namsonx** [comment]
Hello Hogler,

This is just configuration files not a programming language :)
If that, this issue and the issue #79 are duplicated. And I will implement this feature.

Thank you,
Son

**Comment by test-fullautomation** [comment]
Hi Son,
To avoid misunderstanding: The $-Syntax which we introduce with jsonp is a part of a programming language and must behave in any case like a software developer expects.
So, please think big. Any use-case is possible. We don't know how somebody structures json files.
Than you,
 Thomas

**Comment by HolQue** [comment]
Retest successful

**Comment by test-fullautomation** [comment]
solved and already merged with #147
